added file upload to amazon s3

lamuria opened the pull request:
Added file upload to amazon s3 and get file url functions.
With this new functions we are able to upload any kind of files to s3.

Usage:

```
  Inputs: - credentials {:bucket
                         :access-key
                         :secret}
          - file-key We ensure an unique file key by concating an uuid, if nil we generate an unique file-key.
          - file-value {:value ;file-content
                        :content-type }

  Returns: file url to s3

  (file-management/put-file credentials-hash file-key file-info)
```

ref: https://github.com/clanhr/mothership/issues/1266

donbonifacio commented:
Is the ref correct?

:ok: 

Could have a better commit, and also some docstrings on those fns. Also, do we have tests for this?

lamuria commented:
@donbonifacio What kind of tests are you talking about?

donbonifacio commented:
@lamuria the actual send something to s3.

donbonifacio commented on src/file_management/core.clj:
```diff
@@ -48,3 +56,26 @@
+
+(defn gets-file-key
+  [file-key]
+  (str file-key "-" (java.util.UUID/randomUUID)))
```
On your commit message you say that if the file-key exists it will be smashed. Don't this line prevent it?

lamuria replied:
yep! good point! This was done after I wrote the commit message..

lamuria committed:
added file upload to amazon s3

Added file upload to amazon s3 and get file url functions.
With this new functions we are able to upload any kind of files to s3.

Usage:

Inputs: - credentials {:bucket
                       :access-key
                       :secret}
        - file-key We ensure an unique file key by concating an uuid, if nil we generate an unique file-key
        - file-value {:value ;file-content
                      :content-type}

Returns: file url to s3

(file-management/put-file credentials-hash file-key file-info)

ref: https://github.com/clanhr/mothership/issues/1266``

lamuria committed:
Bumpup library version to 0.7.0
